File: desafio_em.java
// Para ler e escrever dados em Java, na "DIO" padronizaram da seguinte maneira:
// 1. (new Scanner(System.in)) - Cria um leitor de entradas, com metodos ulteis com prefixo "next";
// 2. (System.out.println) - Imprime na tela, um texto de saida (Output) e pulando uma linha;


// Realizando o Desafio em Java

import java.util.Scanner;

public class desafio_em {

    public static void main (String[] args) {
        //Lê os valores de Entrada
        Scanner leitorDeEntradas = new Scanner(System.in);
float valorSalario = leitorDeEntradas.nextFloat();
float valorBeneficios = leitorDeEntradas.nextFloat();

float valorImposto =0;
if (valorSalario>= 0 && valorSalario <= 1100){
//Atribuiu a aliquota de 5% mediante salario
valorImposto = 0.05F * valorSalario;
}
 
else if (valorSalario >= 1100.01 && valorSalario <= 2500.00){
    valorImposto = 0.10F * valorSalario;
}
 else {
    valorImposto = 0.15F * valorSalario;
 }


//Calcula e imprime a saída (com 2 casas decimais)
float saida = valorSalario + valorImposto + valorBeneficios;
System.out.println(String.format("%.2f", saida));

// Fecha o scanner
leitorDeEntradas.close();

    }
}

File: desafio_em.js
// Desafios JavaScript na DIO tem funções "gets" e "print" acessiveis globalmente:
// "gets" lê uma linha de dados de entrada (inputs) do usuario;
// "print" Imprime um texto de saida (output), pulando uma linha.

//Lê os valores de entrada:
const valorSalario = parseFloat(gets());
const valorBeneficios = parseFloat(gets());

// Calcula o imposto atraves da função "calcularImposto":
const valorImposto = calcularImposto(valorSalario);

//Calcula e imprime a saida (com 2 casas decimais):
const saida = valorSalario - valorImposto + valorBeneficios;
print(saida.toFixed(2));

// Função util para o calculo do imposto (baseado nas aliquotas)
function calcularImposto(salario) { 
    let aliquota;
    if (salario >= 0 && salario <= 1100) {
        aliquota = 0.5;
    }
     else if (salario >= 1100.01 && salario <= 2500.00) {
        aliquota = 0.10;
     } else  {
       aliquota = 0.15;
     }

    return aliquota * salario;
}
